File: dotnet/src/Experimental/Process.Abstractions/KernelProcessMap.cs
// Copyright (c) Microsoft. All rights reserved.
using System.Collections.Generic;

namespace Microsoft.SemanticKernel;

/// <summary>
/// A serializable representation of a ProcessMap.
/// </summary>
public sealed record KernelProcessMap : KernelProcessStepInfo
{
    /// <summary>
    /// The map operation.
    /// </summary>
    public KernelProcessStepInfo Operation { get; init; }

    /// <summary>
    /// Creates a new instance of the <see cref="KernelProcess"/> class.
    /// </summary>
    /// <param name="state">The process state.</param>
    /// <param name="operation">The map operation.</param>
    /// <param name="edges">The edges for the map.</param>
    public KernelProcessMap(KernelProcessMapState state, KernelProcessStepInfo operation, Dictionary<string, List<KernelProcessEdge>> edges)
        : base(typeof(KernelProcessMap), state, edges)
    {
        Verify.NotNull(operation, nameof(operation));
        Verify.NotNullOrWhiteSpace(state.StepId, $"{nameof(state)}.{nameof(KernelProcessMapState.StepId)}");
        Verify.NotNullOrWhiteSpace(state.RunId, $"{nameof(state)}.{nameof(KernelProcessMapState.RunId)}");

        this.Operation = operation;
    }
}

File: python/semantic_kernel/processes/kernel_process/kernel_process_map.py
from semantic_kernel.processes.kernel_process.kernel_process_edge import KernelProcessEdge
from semantic_kernel.processes.kernel_process.kernel_process_map_state import KernelProcessMapState
from semantic_kernel.processes.kernel_process.kernel_process_step_info import KernelProcessStepInfo


class KernelProcessMap(KernelProcessStepInfo):
    """Information about a map step in a kernel process."""

    operation: KernelProcessStepInfo

    def __init__(
        self, state: KernelProcessMapState, operation: KernelProcessStepInfo, edges: dict[str, list[KernelProcessEdge]]
    ) -> None:
        """Initializes a new instance of the KernelProcessMap class."""
        super().__init__(KernelProcessMap, state, edges)
        self.operation = operation
